rest_framework/throttling.py

```diff
@@ -101,9 +102,26 @@ def parse_rate(self, rate):
         """
         if rate is None:
             return (None, None)
-        num, period = rate.split('/')
-        num_requests = int(num)
-        duration = {'s': 1, 'm': 60, 'h': 3600, 'd': 86400}[period[0]]
+
+        try:
+            num, period = rate.split('/')
+            num_requests = int(num)
+            # Get rate multiplier value if available
+            period_mult, _ = re.split('[s|m|h|d]', period, maxsplit=1)
+            period_char = re.findall('[s|m|h|d]', period)[0]
```

Comment:
Could you use the remainder of the split, and index the first character to determine the `period_char`? e.g., 

```python
period_mult, period_timescale = re.split('[s|m|h|d]', period, maxsplit=1)
period_char = period_timescale[0]
```

Reply:
Also, wouldn't this incorrectly match a period of `1/10foos`?

Reply:
`.split` with the given regex does not preserve the period character (10hour is split to `["10", "our"]`), thus the `.findall`.

Anyway, I'll review this properly again today when I get back to my laptop and try to think of a more elegant solution.

Reply:
I avoided a more complex regex, that's why I ran into this issue :-)

This seems to solve the problem: `re.findall(r'^\d*[s|m|h|d]', period)`:
* If nothing is returned, the rate is not correctly formatted, will raise IndexError
* Otherwise, it will return both the multiplier (if provided) and the first char of time unit